internal/provider/data_subnet.go

```diff
@@ -75,7 +75,7 @@ func dataSubnetRead(d *schema.ResourceData, m interface{}) error {
 		if err := d.Set("dns_servers", res[idx].DNSServers); err != nil {
 			return err
 		}
-		d.SetId(string(res[idx].ID))
+		d.SetId(fmt.Sprintf("%d", res[idx].ID))
```

Comment:
Either way works, but I think `strconv.Itoa` (https://pkg.go.dev/strconv#Itoa) is usually used if you don't have other things to introduce into the string you are formatting

Reply:
Fixed!